plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/M2RepositoryService.java

```diff
@@ -15,26 +15,79 @@
  *******************************************************************************/
 package com.google.cloud.tools.eclipse.appengine.libraries.repository;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
 
 import com.google.cloud.tools.eclipse.appengine.libraries.MavenCoordinates;
+import com.google.cloud.tools.eclipse.util.MavenUtils;
 
 /**
  * Implementation of {@link ILibraryRepositoryService} that relies on M2Eclipse to download the artifacts and store
  * them in the local Maven repository pointed to by M2Eclipse's M2_REPO variable.
  */
-// FIXME For now this class is just a mock, to be implemented soon
+@Component
 public class M2RepositoryService implements ILibraryRepositoryService {
 
   @Override
-  public IPath getJarLocation(MavenCoordinates mavenCoordinates) {
-    return new Path("/path/to/jar/file/in/m2_repo/" + mavenCoordinates.getArtifactId() + "." + mavenCoordinates.getType());
+  public IPath getJarLocation(MavenCoordinates mavenCoordinates) throws LibraryRepositoryServiceException {
+    try {
+      List<ArtifactRepository> repository = getRepository(mavenCoordinates);
+
+      Artifact artifact;
+      if (mavenCoordinates.getVersion().equals(MavenCoordinates.LATEST_VERSION)) {
+        artifact = MavenUtils.resolveLatestReleasedArtifact(null, mavenCoordinates.getGroupId(), mavenCoordinates.getArtifactId(), mavenCoordinates.getType(), mavenCoordinates.getClassifier(), repository);
+      } else {
+        artifact = MavenUtils.resolveArtifact(null, mavenCoordinates.getGroupId(), mavenCoordinates.getArtifactId(), mavenCoordinates.getType(), mavenCoordinates.getVersion(), mavenCoordinates.getClassifier(), repository);
+      }
+
+      return new Path(artifact.getFile().getAbsolutePath());
+    } catch (CoreException ex) {
+      throw new LibraryRepositoryServiceException("Could not resolve maven artifact: " + mavenCoordinates, ex);
+    }
+  }
+
+  private ArtifactRepository getRepository(String repository) throws LibraryRepositoryServiceException {
+    try {
+      URI repoUri = new URI(repository);
+      if (!repoUri.isAbsolute()) {
+        throw new LibraryRepositoryServiceException("repository URI must be an absolute URI (i.e. has to have a "
+            + "schema component)");
```

Comment:
I suggest to add `+ repoUri.toString()` (or something like that) to the exception message.